### PR TITLE
[Kimi-K2.5] E2E Pipeline + Video Support

### PR DESCRIPTION
**Pull request by depksingh**
## Motivation
Adding the support for Kimi-K2.5 E2E pipeline along with enabling video support for the model.

## Modifications

- Code changes related to supporting https://huggingface.co/moonshotai/Kimi-K2.5 
- Added support for video enablement.
- Added adapter for missing `mecord` dependency from Moonshot AI [Discussion Link](https://huggingface.co/moonshotai/Kimi-K2.5/discussions/25)
- Stage config changes for Kimi-K2.5

This PR takes care of the E2E changes, and tested and built on top of https://github.com/sgl-project/sglang-jax/pull/1637 PR, which contains the INT4 changes. To test it E2E you can either use BF16 checkpoint [here](https://huggingface.co/ananayarora/Kimi-K2.5-BF16) with 2x2x4 v7x or merge the INT4 PR with these changes and spin up the server with 2x2x1 v7x.

## Accuracy Tests

MMMU Eval: [Link](https://gist.github.com/chaitanyapk/b07160398746c2c37818506f7993376d)
MMLU Eval: [Link](https://github.com/sgl-project/sglang-jax/pull/1388)

## Benchmarking and Profiling

Server Launch 
```
python3 -m sgl_jax.launch_server  --model-path=moonshotai/Kimi-K2.5 --tp-size 8 --page-size 64 --mem-fraction-static 0.9 --host 0.0.0.0 --port 30000 --trust-remote-code --chunked-prefill-size 16384 --disable-radix-cache --multimodal
```

Text, Image, Video Request Responses: [Link](https://gist.github.com/depksingh/7472e286872e23438750311d7e9eb388)

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [x] The test plan, such as providing test command.
- [x] (Optional) The necessary documentation update.

### CODE REVIEW

**Comment by depksingh** [comment]
Hello @Bob-Chen222 , @cjx0709 , @chaitanyapk  can you please review the PR whenever you get time.

**Comment by Yanko-7** [comment]
Could we migrate Kimi-K2.5 to the VLM architecture that we currently maintain, instead of adding it to the MultiStage pipeline?

For regular VLMs, we would like to standardize on the in-model multimodal path (`InModelMultimodalContract`), similar to `Qwen3VLForConditionalGeneration`. Going forward, MultiStage should be reserved for Omni models, where multiple heterogeneous stages are actually needed.

For Kimi-K2.5, I think the preferred direction would be to:

* integrate the vision tower into `KimiK25ForConditionalGeneration`;
* implement the `InModelMultimodalContract` interfaces, such as `get_input_embeddings()` and `get_multimodal_encode_funcs()`;
* reuse the existing in-model multimodal orchestration for image/video embeddings;

This would keep Kimi-K2.5 consistent with the VLM path we plan to maintain and avoid introducing another VLM dependency on the legacy MultiStage architecture.

**Comment by depksingh** [comment]
Hello @Yanko-7 , Thank you for your review. 
I updated the changes inline with the in-model multimodal orchestration as you suggested. Can you please review the changes.

File: python/sgl_jax/srt/multimodal/models/kimi_k25/kimi_k25_vit.py [hunk 1]
```diff
@@ -255,10 +258,7 @@ def local_flash_attention(q, k, v, segment_ids):
             check_vma=False,
         )(pad_q, pad_k, pad_v, segment_ids)
 
-        output = jnp.transpose(output[0], (1, 0, 2))
-        output = output[:sum_seq_len, :, :].reshape(sum_seq_len, D)
-
-        return output
+        return jnp.transpose(output[0], (1, 0, 2))[:seq_len, :, :]
```
Comment (Yanko-7) [inline python/sgl_jax/srt/multimodal/models/kimi_k25/kimi_k25_vit.py:312]: You don't need to implement this yourself. Please use the wrapper in `sgl_jax/srt/multimodal/layers/attention/flash_attention_backend.py`.
you can call `make_vision_attention_backend` with `use_varlen=True` to use `VisionVarlenAttentionBackend`.

File: python/sgl_jax/srt/multimodal/models/kimi_k25/kimi_k25_vit.py
Comment (Yanko-7) [inline python/sgl_jax/srt/multimodal/models/kimi_k25/kimi_k25_vit.py:1]: Could we introduce `VisionShardSpecs` and run the ViT forward pass through `run_mrope_vision_model`? This would allow us to support both DP-Encoder and TP-Encoder.

And please test whether it can start normally.

**Comment by depksingh** [comment]
Hi @Yanko-7 , Gentle reminder on this

File: python/sgl_jax/srt/multimodal/models/kimi_k25/kimi_k25_vit.py (hunk 1 above)
Comment (depksingh) [reply python/sgl_jax/srt/multimodal/models/kimi_k25/kimi_k25_vit.py]: Thanks for sharing, updated

File: python/sgl_jax/srt/multimodal/models/kimi_k25/kimi_k25_vit.py
Comment (depksingh) [reply python/sgl_jax/srt/multimodal/models/kimi_k25/kimi_k25_vit.py:1]: yes, integrated both the visionShardSpecs and refactored code to forward pass through `run_mrope_vision_model`. Tested all the image, text, video flow, working fine for all the scenarios.

**Comment by depksingh** [comment]
Hello @Yanko-7 , thank you for the review, I've updated the code as per the comments, can you please review it.

**Comment by Yanko-7** [comment]
LGTM overall. I've made a few changes, and there are no blocking issues from my side.Let's wait for #1681 to land first, and then I'll make a few follow-up changes here to align with it.

**Comment by depksingh** [comment]
sure @Yanko-7 , Thank you for taking the time to review.
